tests/unit/python/foglamp/common/test_process.py

```diff
@@ -0,0 +1,102 @@
+# -*- coding: utf-8 -*-
+
+import pytest
+
+from unittest.mock import patch
+
+from foglamp.common.storage_client.storage_client import ReadingsStorageClient, StorageClient
+from foglamp.common.process import FoglampProcess, SilentArgParse, ArgumentParserError
+from foglamp.common.microservice_management_client.microservice_management_client import MicroserviceManagementClient
+
+
+__author__ = "Ashwin Gopalakrishnan"
+__copyright__ = "Copyright (c) 2017 OSIsoft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+
+@pytest.allure.feature("unit")
+@pytest.allure.story("common", "foglamp-process")
+class TestFoglampProcess:
+
+    def test_constructor_abstract_method_run(self):
+        with pytest.raises(TypeError):
+            fp = FoglampProcess()
+        with pytest.raises(TypeError):
+            class FoglampProcessImp(FoglampProcess):
+                pass
+            fp = FoglampProcessImp()
+
+    @pytest.mark.parametrize('argslist',
+                             [([ArgumentParserError()]),
+                              (['corehost', ArgumentParserError()]),
+                              (['corehost', 0, ArgumentParserError()])
```

Comment:
I don;t think this covers the code path; we should test with  combinations of`address, host and port` itself and for error try some bad say address1

Reply:
I am not sure what you mean. I have mocked an error coming from the silent parser. As we are testing application arguments, and the application, in this case, is the unit test process itself, I chose to mock the parser to simulate actual command line arguments.

Reply:
**Please add test for ValueError handling**

```
[None, None, None] > ValueError: --address is not specified

[None, 1, 'a name'] > ValueError: --address is not specified

['corehost', None, 'a name'] > ValueError: --port is not specified

['corehost', 1, None] > ValueError: --name is not specified
```

I think code should have handling of empty string as well, Not only `None`

e.g. `if self._core_management_host is None or self._core_management_host == "":`

Test data `["", 1, "a name"]`

Reply:
I didn't add cases for this as I believe the code would never return None. I didn't want to refactor too much as the issue is about adding basic unit tests without refactoring. I have filed an issue for this improvement but I believe it is okay to proceed as is.